onnxruntime/python/tools/transformers/float16.py

```diff
@@ -458,9 +484,24 @@ def convert_float_to_float16(
                     model.graph.node.extend(new_node)
                     # change current node's input name
                     node.input[i] = output_name
+                    is_input_converted = True
                     break
-        # if output's name is in the value_info_list meaning output is tensor(float16) type, insert a float to
-        # float16 Cast node after the node, change current node's output name and create new value_info for the new name
+
+            # For bfloat16 nodes, we need to convert float initializers to bfloat16.
+            if (not is_input_converted) and use_bf16 and (input_name in initializers_to_be_casted_to_bf16):
+                output_name = node.name + "_input_cast_" + str(i)
+                value_info = helper.make_tensor_value_info(
+                    name=output_name, elem_type=accuracy_type, shape=initializers_to_be_casted_to_bf16[input_name].dims
+                )
+                new_value_info = model.graph.value_info.add()
+                new_value_info.CopyFrom(value_info)
+                node_name = node.name + "_input_cast" + str(i)
+                new_node = [helper.make_node("Cast", [input_name], [output_name], to=accuracy_type, name=node_name)]
```

Comment:
So we should have an initializer that is type of bfloat16, right? Why we need a Cast here? If the initializer is type of bfloat16, then its subsequent operator can match type, then we should not have a Cast. Is that true?

Reply:
Undo the change of this file in PR so that we can focus on graph fusion for this PR. I will add a new PR for bf16 conversion later.